openedx/core/djangoapps/user_api/api.py

```diff
@@ -0,0 +1,814 @@
+
+    """
+    form_desc = FormDescription("post", reverse("password_change_request"))
+
+    # Translators: This label appears above a field on the password reset
+    # form meant to hold the user's email address.
+    email_label = _(u"Email")
+
+    # Translators: This example email address is used as a placeholder in
+    # a field on the password reset form meant to hold the user's email address.
+    email_placeholder = _(u"username@domain.com")
+
+    # Translators: These instructions appear on the password reset form,
+    # immediately below a field meant to hold the user's email address.
+    email_instructions = _(u"The email address you used to register with {platform_name}").format(
+        platform_name=configuration_helpers.get_value('PLATFORM_NAME', settings.PLATFORM_NAME)
+    )
+
+    form_desc.add_field(
+        "email",
+        field_type="email",
+        label=email_label,
+        placeholder=email_placeholder,
+        instructions=email_instructions,
+        restrictions={
+            "min_length": accounts.EMAIL_MIN_LENGTH,
+            "max_length": accounts.EMAIL_MAX_LENGTH,
+        }
+    )
+
+    return form_desc
+
+
+def get_login_session_form():
+    """Return a description of the login form.
+
+    This decouples clients from the API definition:
+    if the API decides to modify the form, clients won't need
+    to be updated.
+
+    See `user_api.helpers.FormDescription` for examples
+    of the JSON-encoded form description.
+
+    Returns:
+        HttpResponse
+
+    """
+    form_desc = FormDescription("post", reverse("user_api_login_session"))
+
+    # Translators: This label appears above a field on the login form
+    # meant to hold the user's email address.
+    email_label = _(u"Email")
+
+    # Translators: This example email address is used as a placeholder in
+    # a field on the login form meant to hold the user's email address.
+    email_placeholder = _(u"username@domain.com")
+
+    # Translators: These instructions appear on the login form, immediately
+    # below a field meant to hold the user's email address.
+    email_instructions = _("The email address you used to register with {platform_name}").format(
+        platform_name=configuration_helpers.get_value('PLATFORM_NAME', settings.PLATFORM_NAME)
+    )
+
+    form_desc.add_field(
+        "email",
+        field_type="email",
+        label=email_label,
+        placeholder=email_placeholder,
+        instructions=email_instructions,
+        restrictions={
+            "min_length": accounts.EMAIL_MIN_LENGTH,
+            "max_length": accounts.EMAIL_MAX_LENGTH,
+        }
+    )
+
+    # Translators: This label appears above a field on the login form
+    # meant to hold the user's password.
+    password_label = _(u"Password")
+
+    form_desc.add_field(
+        "password",
+        label=password_label,
+        field_type="password",
+        restrictions={
+            "min_length": accounts.PASSWORD_MIN_LENGTH,
```

Comment:
I think you should remove `min_length` restriction from login form, see https://openedx.atlassian.net/browse/LEARNER-1209